src/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/PrimsAfterRollback.hpp

```diff
@@ -0,0 +1,71 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Tags/DidRollback.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace EquationsOfState {
+template <bool IsRelativistic, size_t ThermodynamicDim>
+class EquationOfState;
+}  // namespace EquationsOfState
+template <size_t Dim>
+class Mesh;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace grmhd::GhValenciaDivClean::subcell {
+/*!
+ * \brief Mutator that resizes the primitive variables to the subcell mesh and
+ * computes the primitives, but only if
+ * `evolution::dg::subcell::Tags::DidRollback` is `true`.
+ *
+ * In the DG-subcell `step_actions` list this will normally be called using the
+ * `::Actions::MutateApply` action in the following way in the action list:
+ * - `Actions::Label<Labels::BeginSubcellAfterDgRollback>`
+ * - `Actions::MutateApply<PrimsAfterRollback<primitive_recovery_schemes>>`
+ * Note that the GR tags are swapped to the subcell grid before this action is
```

Comment:
Sorry, shouldn't this be omitted as well?

Reply:
Yes... done and squashed :)